### PR TITLE
refactor treebucketer

### LINKED ISSUE #121: Refactor TreeBucketer
Currently we have:

https://github.com/ing-bank/probatus/blob/2a71061e4662316619bd1e731c7c5094e7b707bb/probatus/binning/binning.py#L388-L418

However we could refactor using 

```python
from sklearn.tree import _tree
splits = np.unique(tree.tree_.threshold[tree.tree_.feature != _tree.TREE_UNDEFINED])
```

Which would mean we can remove all of `probatus/utils/tree.py`.

Let me know and I'll open a PR.

### PR DESCRIPTION
**Pull request by timvink**
Closes #121

### CODE REVIEW

**Comment by timvink** [comment]
Btw, I'm enjoying this:

image

File: probatus/binning/binning.py [hunk 1]
```diff
@@ -437,9 +436,8 @@ def tree_bins(x, y, inf_edges, tree):
 
         bin_count = len(index)
 
-        tpf = TreePathFinder(tree)
-        boundaries = [bound["min"] for bound in tpf.get_boundaries().values()]
-        boundaries += [tpf.get_boundaries()[leaves[-1]]["max"]]
+        boundaries = np.unique(tree.tree_.threshold[tree.tree_.feature != _tree.TREE_UNDEFINED])
+        boundaries = [-np.inf] + boundaries.tolist() + [np.inf]
 
         if not inf_edges:
```
Comment (Matgrb) [inline probatus/binning/binning.py:442]: Maybe to be fully consistent with other classes, in line 440 add as boundaries min and max for the data, and then 

```
if inf_edge:
    change the boundaries to infinite
```

File: probatus/binning/binning.py [hunk 2]
```diff
@@ -437,9 +436,8 @@ def tree_bins(x, y, inf_edges, tree):
 
         bin_count = len(index)
 
-        tpf = TreePathFinder(tree)
-        boundaries = [bound["min"] for bound in tpf.get_boundaries().values()]
-        boundaries += [tpf.get_boundaries()[leaves[-1]]["max"]]
+        boundaries = np.unique(tree.tree_.threshold[tree.tree_.feature != _tree.TREE_UNDEFINED])
+        boundaries = [-np.inf] + boundaries.tolist() + [np.inf]
```
Comment (Matgrb) [inline probatus/binning/binning.py:440]: Could you add a very simple unit test for that function:
- simple distribution consisting of 5 samples, where you can calculate where exactly the boundaries should be
- simple tree based model
- assert if the precalculated boundaries are compared to what the function returns?
- assert bin_count  parameter if they have the value correct as you precalculated

The current tests are a bit more general, and i doubt if they would catch a mistake in that line.

File: probatus/binning/binning.py [hunk 3]
```diff
@@ -437,9 +436,8 @@ def tree_bins(x, y, inf_edges, tree):
 
         bin_count = len(index)
 
-        tpf = TreePathFinder(tree)
-        boundaries = [bound["min"] for bound in tpf.get_boundaries().values()]
-        boundaries += [tpf.get_boundaries()[leaves[-1]]["max"]]
```
Comment (Matgrb) [inline probatus/binning/binning.py:442]: In the previous code the boundary was added only on the right side, but in the new code it is added to both left and right side. Could you double check if it is correct?

**Review by Matgrb — CHANGES_REQUESTED** [review]
Great work, indeed it feels great to remove code :)

File: probatus/binning/binning.py (hunk 3 above)
Comment (timvink) [reply probatus/binning/binning.py:442]: I did, it's correct. I think it's some sort of max of the leaves or something

File: probatus/binning/binning.py (hunk 1 above)
Comment (timvink) [reply probatus/binning/binning.py]: done.

File: probatus/binning/binning.py (hunk 2 above)
Comment (timvink) [reply probatus/binning/binning.py]: add unit tests.

**Comment by timvink** [comment]
macos tests fail. I doubt it's due to my changes. will re-run tests later

**Comment by Matgrb** [comment]
I will check it out. Thanks @timvink

**Review by Matgrb — APPROVED** [review]
Great work!

**Pull request merged**